### ISSUE TITLE
error: password authentication failed for user "immich"

### ISSUE THREAD

**Issue by Deses** [imagegenius/docker-immich#467]
For the past days I've been having this issue:
```
[Nest] 4472  - 12/01/2024, 8:47:21 PM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (9)...
error: password authentication failed for user "immich"
    at Parser.parseErrorMessage (/app/immich/server/node_modules/pg-protocol/dist/parser.js:283:98)
    at Parser.handlePacket (/app/immich/server/node_modules/pg-protocol/dist/parser.js:122:29)
    at Parser.parse (/app/immich/server/node_modules/pg-protocol/dist/parser.js:35:38)
    at Socket.<anonymous> (/app/immich/server/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:518:28)
    at addChunk (node:internal/streams/readable:561:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:512:3)
    at Readable.push (node:internal/streams/readable:392:5)
    at TCP.onStreamRead (node:internal/stream_base_commons:191:23)
[Nest] 4472  - 12/01/2024, 8:47:21 PM   ERROR [ExceptionHandler] password authentication failed for user "immich"
error: password authentication failed for user "immich"
    at Parser.parseErrorMessage (/app/immich/server/node_modules/pg-protocol/dist/parser.js:283:98)
    at Parser.handlePacket (/app/immich/server/node_modules/pg-protocol/dist/parser.js:122:29)
    at Parser.parse (/app/immich/server/node_modules/pg-protocol/dist/parser.js:35:38)
    at Socket.<anonymous> (/app/immich/server/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:518:28)
    at addChunk (node:internal/streams/readable:561:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:512:3)
    at Readable.push (node:internal/streams/readable:392:5)
    at TCP.onStreamRead (node:internal/stream_base_commons:191:23)
api worker exited with code 1
Starting api worker
```
Immich doesn't start, and I haven't changed anything for it to break. I even changed the db password for user `immich` to something new, verified the password works through DBeaver, and nothing, Immich still doesn't manage to load.

These are the parameters I'm using to launch the image:
```
docker run
  -d
  --name='immich'
  --net='nirvnet'
  --pids-limit 2048
  -e TZ="Europe/Paris"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="Unraid"
  -e HOST_CONTAINERNAME="immich"
  -e 'DB_HOSTNAME'='postgresql14'
  -e 'DB_USERNAME'='immich'
  -e 'DB_PASSWORD'='aStrongPassword'
  -e 'DB_DATABASE_NAME'='immich'
  -e 'REDIS_HOSTNAME'='redis'
  -e 'DISABLE_MACHINE_LEARNING'='false'
  -e 'DISABLE_TYPESENSE'='false'
  -e 'DB_PORT'='5432'
  -e 'REDIS_PORT'='6379'
  -e 'REDIS_PASSWORD'=''
  -e 'PUID'='99'
  -e 'PGID'='100'
  -e 'UMASK'='022'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:8080]'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/imagegenius/templates/main/unraid/img/immich.png'
  -p '9210:8080/tcp'
  -v '/mnt/user/Fotos/':'/photos':'rw'
  -v '/mnt/user/Fotos/machine-learning':'/config/machine-learning':'rw'
  -v '/mnt/user/appdata/immich':'/config':'rw' 'ghcr.io/imagegenius/immich:latest'
```

This is running on an Unraid environment and the Postgres repository i'm using is `tensorchord/pgvecto-rs:pg14-v0.2.0`, while Redis is just `redis`.

**Comment by hydazz** [comment]
Do you have multiple postgres14 instances? perhaps its getting confused with `postgresql14` - try using ip address?

**Comment by Deses** [comment]
Nope, I just have one but I'll try that.

**Comment by Deses** [comment]
Oh I looked at the logs of my postgres DB and it was full of this:

```JS
2024-12-01 20:48:27.527 CET [23495] FATAL:  password authentication failed for user "immich"
2024-12-01 20:48:27.527 CET [23495] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
2024-12-01 20:48:30.539 CET [23496] FATAL:  password authentication failed for user "immich"
2024-12-01 20:48:30.539 CET [23496] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
2024-12-01 20:48:33.546 CET [23497] FATAL:  password authentication failed for user "immich"
2024-12-01 20:48:33.546 CET [23497] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
2024-12-01 20:48:36.555 CET [23498] FATAL:  password authentication failed for user "immich"
2024-12-01 20:48:36.555 CET [23498] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
```

**Comment by Deses** [comment]
Well, I figured it out. It was a problem with the password itself.

I was using this exact one `R]44JM3t{<%^i$:8D7!f}BwMdaMmJb[:WDH\0P`, which I assure you it was working for YEARS just fine and I could connect with it through database admin tools like DBeaver.

Probably because that \ character or maybe because the %, it no longer liked that password so I made a simpler one without _special_ special characters and now it finally works.
